doc/common/Cadl-Project-Scripts.md

```diff
@@ -0,0 +1,127 @@
+# Integrating with cadl sync and generate scripts
+
+There are 2 common scripts provided for each language to be able to generate from within the language
+repo and use the remote cadl definition in the spec repo.
+
+## Prerequisites
+
+There are 4 things that these two scripts expect are set up in your language repo before they will run correctly.
+
+### cadl-location.yaml
+
+This file should live under the project directory for each service and has the following properties
+
+| Property | Description | IsRequired |
+| --- | --- | --- |
+| directory | The top level directory where the main.cadl for the service lives.  This should be relative to the spec repo root such as `specification/cognitiveservices/OpenAI.Inference` | true |
+| additionalDirectories | Sometimes a cadl file will use a relative import that might not be under the main directory.  In this case you can specify additional directories as a list to sync so that all needed files are synced. | false: default = null |
+| commit | The commit sha for the version of the cadl files you want to generate off of.  This allows us to have idempotence on generation until we opt into pointing at a later version. | true |
+| repo | The repo this spec lives in.  This should be either `Azure/azure-rest-api-specs` or `Azure/azure-rest-api-specs-pr`.  Note that pr will work locally but not in CI until we add another change to handle token based auth. | true |
+| cleanup | This will remove the TempCadlFiles directory after generation is complete if true otherwise this directory will be left to support local changes to the files to see how different changes would affect the generation. | false: default = true |
+
+Example
+
+```yml
+directory: specification/cognitiveservices/OpenAI.Inference
+additionalDirectories:
+  - specification/cognitiveservices/OpenAI.Authoring
+commit: 14f11cab735354c3e253045f7fbd2f1b9f90f7ca
+repo: Azure/azure-rest-api-specs
+cleanup: false
+```
+
+### TempCadlFiles
+
+You should add a new entry in your .gitignore for your repo so that none of these files are accidentally checked in if cleanup is turned off.
+`TempCadlFiles/`
+
+### emitter-package.json
+
+This will be the package.json that gets used when `npm install` is called.  This replaces the package.json checked into the spec repo and allows each language to fix the version of their emitter to be the same for all packages in their repo.
+The file should be checked into this location `./eng/emitter-package.json`
```

Comment:
nit: Should the file name be called `cadl-emitter-package.json` since this is in the SDK repo and having `cadl` in the name gives better context?

Reply:
given this name might change soon i'd prefer to leave it as is.